src/modules/ekf2/ekf2_main.cpp

```diff
@@ -814,25 +825,29 @@ void Ekf2::run()
 			_ekf.setGpsData(gps.timestamp, &gps_msg);
 		}
 
-		// only set airspeed data if condition for airspeed fusion are met
-		bool fuse_airspeed = airspeed_updated && !vehicle_status.is_rotary_wing
-				     && (_arspFusionThreshold.get() > FLT_EPSILON)
-				     && (airspeed.true_airspeed_m_s > _arspFusionThreshold.get());
+		bool inhibit_use = false;
 
-		if (fuse_airspeed) {
-			const float eas2tas = airspeed.true_airspeed_m_s / airspeed.indicated_airspeed_m_s;
-			_ekf.setAirspeedData(airspeed.timestamp, airspeed.true_airspeed_m_s, eas2tas);
-		}
+		if (!vehicle_status.is_rotary_wing && (_arspFusionThreshold.get() > FLT_EPSILON)) {
+			// only set airspeed data if condition for airspeed fusion are met
+			bool fuse_airspeed = airspeed_updated && (airspeed.true_airspeed_m_s > _arspFusionThreshold.get());
 
-		if (vehicle_status_updated) {
-			// only fuse synthetic sideslip measurements if conditions are met
-			bool fuse_beta = !vehicle_status.is_rotary_wing && (_fuseBeta.get() == 1);
-			_ekf.set_fuse_beta_flag(fuse_beta);
+			if (fuse_airspeed) {
+				const float eas2tas = airspeed.true_airspeed_m_s / airspeed.indicated_airspeed_m_s;
```

Comment:
Constrain or protect this in some way?

Reply:
it is already constrained everywhere it is used inside ecl

Reply:
Ok, it's probably fine then. Although it would be slightly better to do the operation once on the input (@ 80 Hz or so) rather than multiple times in the EKF.